### PR TITLE
Tracie v2

### PR DESCRIPTION
**Pull request by yakazimir**
newest version of tracie.

### CODE REVIEW

**Comment by aimichal** [comment]
Does the data dir need to be called data_public, or can it be renamed to simply data/?

The prediction format is very simple -- one prediction per line, corresponding to each question.

I wonder if it'd be easier if the questions had identifiers (as simple as "instance001", etc.), and the predictions were also JSONL files, with items like

```js
{"id":"instance001", "prediction":"entailment"}
{"id":"instance002", "prediction":"entailment"}
{"id":"instance003", "prediction":"entailment"}
...
```

If you already have code that uses/produces this prediction format, then it's not worth considering. Just a thought1

File: tracie/data_public/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  "story": "Tom needed to get braces. He was afraid of them. The dentist assured him everything would be fine. Tom had them on for a while. Once removed he felt it was worth it.",
+  "gold_label": "contradiction"
+}
+```
+
+and consists of three fields:
+
+* `query` (or hypothesis)
+* `story` (or premise)
+* `gold_label` (the inference label; in `test.jsonl` this is hidden and replaced with `"-"`)
```
Comment (aimichal) [inline tracie/data_public/README.md]: Is this necessary? Can the field simply be skipped, and not given the value "-"?

**Comment by yakazimir** [comment]
> Does the data dir need to be called data_public, or can it be renamed to simply data/?
> 
> The prediction format is very simple -- one prediction per line, corresponding to each question.
> 
> I wonder if it'd be easier if the questions had identifiers (as simple as "instance001", etc.), and the predictions were also JSONL files, with items like
> 
> ```js
> {"id":"instance001", "prediction":"entailment"}
> {"id":"instance002", "prediction":"entailment"}
> {"id":"instance003", "prediction":"entailment"}
> ...
> ```
> 
> If you already have code that uses/produces this prediction format, then it's not worth considering. Just a thought1

Doesn't need to be `data_public`, I at some point saw that and thought it was a convention of some kind. `data/` is just fine. Yeh, about the id point, that's fine, I think it makes sense (I don't mind either way).

**Comment by aimichal** [comment]
@yakazimir More changes:

* Renamed data_public to data
* Removed the labels from the test file (instead of using "-")
* Added ids to each of the files (like `tracie-test-0000`, `tracie-train-iid-0000`, and `tracie-train-uniform-0000`)
* Made the prediction file be a jsonl file, instead of a one-label-per-line text file.

The result is that it's now harder to mismatch a prediction file and an answer file (you'll get errors like `Prediction for id tracie-train-iid-0000 missing` or `Prediction count (1174) doesn't match answer count (860)`).

Also, you can more reasonably compare prediction files, if you are comparing two model predictions, because you can match on the id.

Please have a look, and confirm that things are still running for you. (The scores returned by the evaluator are unchanged; but I used the dummy prediction file. If you use a real model, you should confirm you're still getting the numbers you expect.)

**Comment by yakazimir** [comment]
Looks fine to me. @Slash0BZ is this okay with you (perhaps you want to sanity check on a real output file)

File: tracie/data_public/README.md (hunk 1 above)
Comment (yakazimir) [reply tracie/data_public/README.md]: it can be skipped (it's consistent with a convention from some standard dataset).

**Comment by yakazimir** [comment]
okay, I checked with @Slash0BZ in a separate message, this looks all good to us.

**Pull request merged**